components/common-go/util/workspaceid.go

```diff
@@ -0,0 +1,465 @@
+// Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package util
+
+import (
+	"math/rand"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// UnmarshalJSON parses the duration to a time.Duration
+func GenerateWorkspaceID() string {
+	return random(colors, 1) + "-" + random(animals, 1) + "-" + random(characters, 8)
+}
+
+var WorkspaceIdPattern = regexp.MustCompile(`^[a-z]{3,12}-[a-z]{2,16}-[a-z0-9]{8}$`)
+
+func random(array []string, length int) string {
+	rand.Seed(time.Now().UnixNano())
+	var result = ""
+	for i := 0; i < length; i++ {
+		result += array[rand.Intn(len(array))]
+	}
+	return result
+}
+
+var characters = strings.Split("abcdefghijklmnopqrstuvwxyz0123456789", "")
+
+var colors = []string{
```

Comment:
Considering we're maintaining this list in two places, I wonder if would make sense to keep this as txt file and use something like `go generate`/`embed`/go.rice in Go, load the text file on startup in TS.

This way we'd prevent those lists from drifting apart over time.

Reply:
I don't expect changes in those lists. So maybe this is something we should invest in once we start changing those lists (or drift becomes a burden).